pkg/controllers/cloud/networkpolicy.go

```diff
@@ -1151,6 +1157,37 @@ func (a *appliedToSecurityGroup) notifyNetworkPolicyChange(r *NetworkPolicyRecon
 	}
 }
 
+func (a *appliedToSecurityGroup) removeStaleMembers(stales []*types.NamespacedName, r *NetworkPolicyReconciler) {
+	if len(a.members) == 0 {
+		return
+	}
+	srcMap := make(map[string]*securitygroup.CloudResource)
+	for _, m := range a.members {
+		srcMap[m.Name] = m
+	}
+	for _, stale := range stales {
+		for k := range srcMap {
+			if strings.Contains(stale.Name, k) {
+				r.Log.V(1).Info("Remove stale members from SecurityGroup", "Stale", stale, "Name", a.id.Name)
+				if tracker := r.getCloudResourceNPTracker(srcMap[k], false); tracker != nil {
+					_ = tracker.update(a, true, r)
+				}
+				vmNamespacedName := types.NamespacedName{Name: srcMap[k].Name, Namespace: stale.Namespace}
+				if obj, found, _ := r.virtualMachinePolicyIndexer.GetByKey(vmNamespacedName.String()); found {
+					r.Log.V(1).Info("Delete ANP status", "resource", vmNamespacedName.String())
```

Comment:
shouldnt this be Deleting VMP status for resource vm-id?

Reply:
We are using vm namespaced name in cloud resouce too. Not using vm id because there might be multiple vm CRs for the same instance in different namespaces.